test/hotspot/jtreg/runtime/ErrorHandling/TestHeapDumpOnOutOfMemoryError.java

```diff
@@ -96,11 +97,28 @@ static void test(String type) throws Exception {
         output.stdoutShouldNotBeEmpty();
         output.shouldContain("Dumping heap to " + type + ".hprof");
         File dump = new File(heapdumpFilename);
-        Asserts.assertTrue(dump.exists() && dump.isFile(),
-                "Could not find dump file " + dump.getAbsolutePath());
+        Asserts.assertTrue(dump.exists() && dump.isFile(), "Could not find dump file " + dump.getAbsolutePath());
 
         HprofParser.parse(new File(heapdumpFilename));
-        System.out.println("PASSED");
+
+        // Test again using %p pid substitution in HeapDumpPath:
+        heapdumpFilename = type + ".%p.hprof";
+        pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:+HeapDumpOnOutOfMemoryError",
+                "-XX:HeapDumpPath=" + heapdumpFilename,
+                "-XX:MaxMetaspaceSize=16m",
+                "-Xmx128m",
+                Platform.isDebugBuild() ? "-XX:-VerifyDependencies" : "-Dx",
+                TestHeapDumpOnOutOfMemoryError.class.getName(), type);
+
+        Process proc = pb.start();
```

Comment:
No need differ from the above code here. You can just use OutputAnalyzer.pid() to get the pid in the code below.